python/lsst/ts/wep/task/calcZernikesNeuralTask.py

```diff
@@ -0,0 +1,1906 @@
+            for i in range(numStamps):
+                centers = tartsInternalData[i].get('centers', None)
+                if centers is not None:
+                    # Convert tensor to numpy and extract coordinates
+                    centersNp = self._toNumpy(centers)
+                    if len(centersNp) >= 2:
+                        centXList.append(float(centersNp[0]))
+                        centYList.append(float(centersNp[1]))
+                    else:
+                        centXList.append(0.0)
+                        centYList.append(0.0)
+                else:
+                    centXList.append(0.0)
+                    centYList.append(0.0)
+            self.log.debug("Using TARTS internal data for %d centroid positions", numStamps)
+        else:
+            # Use default values when TARTS internal data not available
+            self.log.warning("TARTS internal data not available, using default centroid positions")
+            centXList = [0.0] * numStamps
+            centYList = [0.0] * numStamps
+
+        # Create metadata for all donuts
+        metadata = PropertyList()
+        metadata["RA_DEG"] = [0.0] * numStamps  # Will be calculated from WCS
+        metadata["DEC_DEG"] = [0.0] * numStamps  # Will be calculated from WCS
+        metadata["CENT_X"] = centXList
+        metadata["CENT_Y"] = centYList
+        metadata["DET_NAME"] = [detectorName] * numStamps
+        metadata["CAM_NAME"] = [cameraName] * numStamps
+        metadata["DFC_TYPE"] = [defocalType] * numStamps
+        # Set defocal distance based on focal position
+        # LSST typically uses different defocal distances for intra vs extra
+        defocalDistance = -1.5 if defocalType == "intra" else 1.5  # TODO: Verify actual values
+        metadata["DFC_DIST"] = [defocalDistance] * numStamps
+        metadata["BANDPASS"] = [bandLabel] * numStamps
+        metadata["BLEND_CX"] = ["nan"] * numStamps
+        metadata["BLEND_CY"] = ["nan"] * numStamps
+
+        # Add TARTS-specific metadata - use internal data when available
+        if tartsInternalData and len(tartsInternalData) >= numStamps:
+            # Use TARTS internal data for field positions
+            fxList = []
+            fyList = []
+            for i in range(numStamps):
+                fxList.append(float(tartsInternalData[i].get('fx', 0.0)))
+                fyList.append(float(tartsInternalData[i].get('fy', 0.0)))
+            metadata["FX"] = fxList
+            metadata["FY"] = fyList
+            self.log.debug("Using TARTS internal data for field positions")
+        else:
+            # Use default values when TARTS internal data not available
+            self.log.warning("TARTS internal data not available, using default field positions")
+            metadata["FX"] = [0.0] * numStamps
+            metadata["FY"] = [0.0] * numStamps
+
+        # Create list of DonutStamp objects
+        donutStamps = []
+
+        for i in range(numStamps):
```

Comment:
in here as well, I really don't see why can't you do all this handling in the previous if staement, you are getting the centers here again, but you got them before as well

Reply:
Is this code going to be moved like Guillem said?

Reply:
Looks like this still hasn't been done.